Architecture.md

```diff
@@ -73,10 +95,49 @@
 
 #### Client Applications
 
-***TODO: [INFRA] how are the Client applications provisioned, how they are exposed and how the underlying infrastructure is managed***
+Pomegranate is responsible for starting, stopping, and interacting with the client applications.
+It is also responsible for managing the networking of the applications.
+
+Pomegranate is completely stateless, and the API is the only external way to interact with it.
+Once a container is started, Pomegranate does not keep track of it, and it is the responsibility of the execution engine to manage it.
+
+```mermaid
+flowchart TD
+    web[80]
+    webscure[443]
+    traefik[Traefik]
+    app_1[User app 1]
+    app_2[User app 2]
+    app_3[User app 3]
+
+    rules{Resolve\nvia labels}
+
+    webscure --> |HTTPS| traefik
+    web --> |HTTP| traefik
+    traefik --> rules
+
+    rules --> |uuid_1.user-app:80| app_1
+    rules --> |uuid_2.user-app:80| app_2
+    rules --> |uuid_3.user-app:80| app_3
+
+```
+
+To expose newly spawned containers to the World Wide Web, [Traefik](https://doc.traefik.io/traefik/) is configured to answer on 80 and 443 ports.
+Then a set of labels is attributed to each container to create a unique subdomain `<app_uuid>.user-app.<fqdn>` redirecting to the port 80 of the associated app.
```

Comment:
Define who handles or transmits the FQDN to Pomegranate. If not given to Pomegranate, where should it fit and what is it?

Reply:
It is given to Pomegranate through an env variable, is this really relevant to state ?
The reason why `<fqdn>` was used is for agnosticism of the environment where it is deployed.

Reply:
I agree with the agnosticism, but the FQDN lacks a proper definition and leaves the reader scratching their head. Could this be improved with a quick run down of how agnosticism has been applied to Promegranate, and how the FQDN allows it to be versatile environment-wise?

Reply:
Should be done